### DISCUSSION TITLE
Does the script work with multiple artboards?

### DISCUSSION THREAD

**Discussion by hittypitty** [joonaspaakko/Batch-Mockup-Smart-Object-Replacement-photoshop-script#3] [Q&A] [answered]
I'm having 2 mockups of a tshirt, one is from front side and another to back. I want to place a same image on front and back. Front and Back are not in the same artboard and are to be exported individually. Can this be done ?

**Comment by joonaspaakko** [comment, marked as answer]
Technically no... but you can if you convert artboards to individual psd files: `File > Export > Artboards to files...`. 

So if the starting point is **`Mockup.psd`**:

```js
// Layers panel
- Artboard 1
    - @front
- Artboard 2
    - @back
```

And you do `File > Export > Artboards to files...`, you'll end up with files:

```md
- Mockup Artboard 1.psd
- Mockup Artboard 2.psd
```

...and the script would have to be along the lines of:

> Any specific settings in this example are irrelevant. 
>
> The only point I'm making with the example code is that when you split artboards into multiple mockup psd files, you then handle them just like any separate mockup files.

```js
#include "Batch Mockup Smart Object Replacement.jsx" 

var output = {
  path: '$/output', 
  format: 'jpg',
  folders: true,
};

mockups([
  
  {
    output: output,
    mockupPath: '$/Mockup Artboard 1.psd', 
    smartObjects: [
      {
        target: '@front',
        input: '$/input',
      },
    ]
  },
  
  {
    output: output,
    mockupPath: '$/Mockup Artboard 2.psd', 
    smartObjects: [
      {
        target: '@back',
        input: '$/input',
      },
    ]
  },
  
]);
```

**Reply by hittypitty** [reply]
Thankyou so much. I'll try that out.
